pkg/internal/authorizer/authorizer.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorizer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/gobuffalo/flect"
+	"google.golang.org/grpc/metadata"
+	authv1 "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+const (
+	userHeaderKey = "user"
+)
+
+type AuthorizationClient struct {
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+	client.Client
+}
+
+func (a AuthorizationClient) Get(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+	decision, err := a.authorize(ctx, obj, GetAuthorization{
+		Name:      key.Name,
+		Namespace: key.Namespace,
+	})
+	if err != nil {
+		return fmt.Errorf("authorizing request: %w", err)
+	}
+	if decision != DecisionAllowed {
+		return fmt.Errorf("not allowed: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("forbidden: %w", err)
```

in line with HTTP 403

Reply:
hmm, how about 401?